elastica/contact_forces.py

```diff
@@ -725,3 +843,148 @@ def apply_contact(
             self.k,
             self.nu,
         )
+
+
+class RodMeshSurfaceContactWithGridMethod(NoContact):
+    """
+    This class is for applying contact forces between rod-mesh_surface.
+    First system is always rod and second system is always mesh_surface.
+
+    Examples
+    --------
+    How to define contact between rod and mesh_surface.
+
+    >>> simulator.detect_contact_between(rod, mesh_surface).using(
+    ...    RodMeshSurfaceContactWithGridMethod,
+    ...    k=1e4,
+    ...    nu=10,
+    ...    surface_tol=1e-2,
+    ... )
+    """
+
+    def __init__(
+        self, k: float, nu: float, faces_grid: dict, grid_size: float, surface_tol=1e-4
+    ):
+        """
+
+        Parameters
+        ----------
+        k: float
+            Stiffness coefficient between the plane and the rod-like object.
+        nu: float
+            Dissipation coefficient between the plane and the rod-like object.
+        faces_grid: dict
+            Dictionary containing the grid information of the mesh surface.
+        grid_size: float
+            Grid size of the mesh surface.
+        surface_tol: float
+            Penetration tolerance between the surface and the rod-like object.
+
+        """
+        super(RodMeshSurfaceContactWithGridMethod, self).__init__()
+        # n_faces = faces.shape[-1]
+        self.k = k
+        self.nu = nu
+        self.faces_grid = faces_grid
+        self.grid_size = grid_size
```

Comment:
instead of requesting grid_size as input for the class, include it in the faces_grid "metadata". I explain it later in a different comment on the faces_grid generation. This should be something like self.grid_size = faces_grid["grid_size"]